redis/src/cluster_async/mod.rs

```diff
@@ -867,17 +868,34 @@ where
             .await
             .map(|(result, _)| result),
             Some(ResponsePolicy::OneSucceededNonEmpty) => {
-                future::select_ok(receivers.into_iter().map(|(_, receiver)| {
-                    Box::pin(async move {
-                        let result = convert_result(receiver.await)?;
-                        match result {
-                            Value::Nil => Err((ErrorKind::ResponseError, "no value found").into()),
-                            _ => Ok(result),
-                        }
-                    })
-                }))
-                .await
-                .map(|(result, _)| result)
+                // Try to get the first result that isn't nil or an error. If no such response is found,
+                // return nil if at least one of the servers returned nil; otherwise, return the last received error.
+                let mut futures = receivers
+                    .into_iter()
+                    .map(get_receiver)
+                    .collect::<FuturesUnordered<_>>();
+
+                let mut nil_found = false;
+                let mut last_err = None;
+                while let Some(result) = futures.next().await {
+                    match result {
+                        Ok(Value::Nil) => nil_found = true,
+                        Ok(val) => return Ok(val),
+                        Err(e) => last_err = Some(e),
```

Comment:
I think if an error found you should return it right here.
If GLIDE receives 2 `nil`s and 1 error, it is not propagated to the user.

Reply:
Your suggestion will return error also if we have a valid key to return, for example if we get [nil, error, key].
Committing a fix

Reply:
Hm, right. What is better to return in such case? Error or a key?

Reply:
key